projects/npm-tools/packages/npm-scripts/contrib/prettier/prettier.js

```diff
@@ -1106,31 +1133,57 @@ function prepare(filepath) {
 			file = fallback;
 		}
 	}
-	else {
-		root = getFrontendProjectsRoot(filepath);
 
-		if (root) {
-			const scripts = path.join(
-				root,
-				'projects/npm-tools/packages/npm-scripts'
-			);
+	const portalRoot = getPortalRoot(filepath);
+
+	if (portalRoot) {
+		const modules = path.join(portalRoot, 'modules/node_modules');
 
-			const wrapper = path.join(scripts, 'src/scripts/prettier.js');
+		let scripts = path.join(modules, '@liferay/npm-scripts');
+
+		let wrapper = path.join(scripts, 'src/scripts/prettier.js');
 
-			if (fs.existsSync(wrapper)) {
-				dir = scripts;
-				file = wrapper;
-			}
+		if (!fs.existsSync(wrapper)) {
+			scripts = path.join(modules, 'liferay-npm-scripts');
+
+			wrapper = path.join(scripts, 'src/scripts/prettier.js');
 		}
-		else {
-			const extension = require('vscode').extensions.getExtension(
-				'esbenp.prettier-vscode'
-			);
-
-			if (extension) {
-				dir = path.join(extension.extensionPath);
-				file = path.join(dir, 'node_modules/prettier');
-			}
+
+		const fallback = path.join(modules, 'prettier/bin-prettier.js');
+
+		if (fs.existsSync(wrapper)) {
+			dir = scripts;
+			file = wrapper;
+		}
+		else if (fs.existsSync(fallback)) {
+			dir = modules;
+			file = fallback;
+		}
+	}
```

Comment:
This logic looks to be the exact same as the logic above, one for portalRoot and one for workspaceRoot. Can we only keep only block of this logic and instead do

```js
const portalOrWorkspaceRoot = getPortalRoot(filepath) || getWorkspaceRoot(filepath);

if (portalOrWorkspaceRoot) {
//...
```